components/snackbar.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+  <v-snackbar v-model="open" :timeout="3000" :color="variant">
+    <span :style="{ color: textColor }">{{ message }}</span>
+    <v-btn :style="{ color: textColor }" text @click="open = false"
+      >Close</v-btn
+    >
+  </v-snackbar>
+</template>
+
+<script lang="ts">
+export default {
+  data() {
+    return {
+      open: false,
+      variant: '',
+      message: '',
+    }
+  },
+  computed: {
+    textColor: function () {
+      const variantToTextColorMap = {
+        success: '#082218',
+        warning: '#2A150D',
+        error: '#FFF8F8',
+        info: '#050709',
+      }
+      return variantToTextColorMap[this.variant]
+    },
+  },
+  created: function () {
+    this.$store.subscribe((mutation, state) => {
+      if (mutation.type === 'snackbar/showSnackbar') {
+        const message = state.snackbar.message
+        const variant = isAcceptedVariant(state.snackbar.variant)
```

Comment:
The `isAcceptedVariant` check might not be entirely necessary, but I normally use Typescript so this is my "OMG this is normal js" type paranoia kicking in I think 😅

Reply:
Can't hurt to have that extra check, I don't mind it personally.